src/news_type.h

```diff
@@ -115,6 +115,10 @@ struct NewsTypeData {
 	NewsDisplay GetDisplay() const;
 };
 
+/** Container for any custom data that must be deleted after the news item has reached end-of-life. */
+struct NewsAllocatedData {};
```

Comment:
Is there a way we can force this type to never be used directly? (Which is presumably undesirable?)

Reply:
We could define all "default" constructors as protected and then you must make a sub class of it. On the other hand, why make a new instance of this when you can pass nullptr as easily?
So I'm not quite sure whether all the guarding against such a thing is worth it. I just added it for a tiny bit more type safety.